pkg/components/state/registry.go

```diff
@@ -35,6 +38,13 @@ var DefaultRegistry *Registry = NewRegistry()
 func NewRegistry() *Registry {
 	return &Registry{
 		stateStores: map[string]func(logger.Logger) state.Store{},
+		verSet: map[string]components.Versioning{
```

Comment:
Can this be done in the code that registers the components? in this case, `cmd/daprd/components/state_etcd.go`?

Two reasons:

1. Keeping all component registration code tidy and in the same place
2. The etcd component may not be registered depending on the build tags and custom configurations

Reply:
But this, do you mean that when a component is registered, it also passes version information? I.e. Add a new `RegisterComponent` func which accepts version mapping.

Reply:
Yes. I am not sure how to implement this in practice, i.e. what the API should look like... But ideally we should keep all information on a component self-contained in the file in `cmd/daprd/components`, rather than in multiple places

Reply:
No worries! I'll have a go at that. 

I agree with the reasoning for sure.